src/IECoreHoudini/DetailSplitter.cpp

```diff
@@ -78,6 +81,77 @@ std::string normalisePath(const std::string& str)
 	return cleanedPath;
 }
 
+static const UT_String tagGroupPrefix( "ieTag_" );
+
+void processTagAttributes( Primitive &primitive )
+{
+	// std::regex is broken in gcc 4.8.x and this regex fails to match correctly, we use boost to avoid the problem for now.
+	boost::regex tagGroupEx(
+		FromHoudiniGeometryConverter::groupPrimVarPrefix().string() + tagGroupPrefix.c_str() + "(.+)"
+	);
+
+	std::set<SceneInterface::Name> uniqueTags;
+
+	auto primVarIt = primitive.variables.begin();
+	while( primVarIt != primitive.variables.end() )
+	{
+		boost::smatch sm;
+		if( boost::regex_match( primVarIt->first, sm, tagGroupEx ) )
+		{
+			PrimitiveVariable::IndexedView<bool> view( primVarIt->second );
+
+			for( auto b : view )
+			{
+				if( b )
+				{
+					uniqueTags.insert(
+						SceneInterface::Name(
+							boost::algorithm::replace_all_copy(
+								std::string( sm[1] ),
+								std::string( "_" ),
+								std::string( ":" )
+							)
+						)
+					);
+					continue;
+				}
+			}
+
+			primVarIt = primitive.variables.erase( primVarIt );
+		}
+		else
+		{
+			primVarIt++;
+		}
+	}
+
+	auto tags = new IECore::InternedStringVectorData();
+	for( const auto &tag : uniqueTags )
+	{
+		tags->writable().push_back( tag );
+	}
+
+	primitive.blindData()->writable()[IECore::InternedString( "tags" )] = tags;
```

Comment:
The key here doesn't need to be an InternedString, so I assume thats a performance thing. If that's the case, lets precontruct it as g_tags as well.